pygdf/dataframe.py

```diff
@@ -318,6 +320,35 @@ def copy(self):
         df._cols = self._cols.copy()
         return df
 
+    def _sanitize_columns(self, col):
+        """Sanitize pre-appended
+           col values
+        """
+        if len(self) == 0 and len(self.columns) > 0:
+            series = Series(col)
+            val = np.empty(len(series.index)) * np.NaN
+            for name in self._cols:
+                self._cols[name] = self._prepare_series_for_add(val)
+
+        col = self._sanitize_values(col)
+        return col
+
+    def _sanitize_values(self, col):
+        """Sanitize col values before
+           being added
+        """
+        index = self._index
+        series = Series(col)
+        if len(series.index) == 0:
```

Comment:
I don't think this is needed.  As noted in https://github.com/gpuopenanalytics/pygdf/issues/123#issuecomment-378298439, adding an empty column to a non-empty frame is invalid even in pandas.